src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/utils.ts

```diff
@@ -0,0 +1,72 @@
+import { TFunction } from 'react-i18next';
+
+import { produceVMDisks } from '@catalog/utils/WizardVMContext';
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import {
+  V1beta1DataVolumeSpec,
+  V1ContainerDiskSource,
+  V1VirtualMachine,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { CDI_BIND_REQUESTED_ANNOTATION } from '@kubevirt-utils/hooks/useCDIUpload/consts';
+import { getTemplateContainerDisks } from '@kubevirt-utils/resources/template';
+import { getDisks, getVolumes } from '@kubevirt-utils/resources/vm';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+
+export const getRegistryHelperText = (template: V1Template, t: TFunction) => {
+  const containerDisks = getTemplateContainerDisks(template);
+
+  if (containerDisks && containerDisks.length > 0)
+    return t('Enter a container image (for example: {{containerDisk}})', {
+      containerDisk: containerDisks[0],
+    });
+};
+
+export const getDiskSource = (
+  vm: V1VirtualMachine,
+  diskName: string,
+): undefined | V1beta1DataVolumeSpec | V1ContainerDiskSource => {
+  if (!diskName) return;
+
+  const installationCD = getDisks(vm)?.find((disk) => disk.name === diskName);
+
+  if (!installationCD) return;
+
+  const cdVolume = getVolumes(vm)?.find((volume) => volume.name === installationCD.name);
+
+  if (!cdVolume) return;
+
+  if (cdVolume.containerDisk) {
+    return cdVolume.containerDisk;
+  }
+
+  if (cdVolume.dataVolume) {
+    const dataVolumeTemplate = vm.spec?.dataVolumeTemplates?.find(
+      (template) => template.metadata?.name === cdVolume.dataVolume.name,
+    );
+
+    if (!dataVolumeTemplate) return;
```

Comment:
can remove, if dataVolumeTemplate.spec don't exisit it will return undefined, simple add ?.spec